### PR TITLE
[CSharp] Additional data extensions for sink models

### PR DESCRIPTION
**Pull request by GeekMasher**
I added support for `logging`, `ldap`, `command-injection`, and `url-redirect` sinks for the data extensions. The names come from the list in the [Java External Flow](https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll#L276-L281).

*Edit:*

Updated names to match https://github.com/github/codeql/pull/12916 : 

- `command-injection`
- `ldap-injection`
- `log-injection`
- `url-redirection`

### CODE REVIEW

**Comment by jcogs33** [comment]
> The names come from the list in the [Java External Flow](https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll#L276-L281).

Would you mind updating `logging` to `log-injection`, `ldap` to `ldap-injection` and `url-redirect` to `url-redirection` to align with the updates to the Java sink kind names which will be merged in https://github.com/github/codeql/pull/12916? 🙏 
 Apologies for the hassle! 🙈

**Comment by michaelnebel** [comment]
> > The names come from the list in the [Java External Flow](https://github.com/github/codeql/blob/main/java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll#L276-L281).
> 
> Would you mind updating `logging` to `log-injection`, `ldap` to `ldap-injection` and `url-redirect` to `url-redirection` to align with the updates to the Java sink kind names which will be merged in #12916? 🙏 Apologies for the hassle! 🙈

Sorry for my duplicate request via slack :-)
Also could you update the sink kind validation in `ExternalFlow.qll`.
Thank you!

**Comment by GeekMasher** [comment]
@jcogs33 @michaelnebel Makes complete sense and happy we are following the same names across the board. My changes are pushed now

**Comment by michaelnebel** [comment]
> @jcogs33 @michaelnebel Makes complete sense and happy we are following the same names across the board. My changes are pushed now

Could you also adjust the change notes and update the validation in `ExternalFlow.qll`?

File: csharp/ql/lib/change-notes/2023-05-09-models-as-data.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* Additional support for `logging`, `ldap`, `command-injection`, and `url-redirect` sink kinds for Models as Data.
```
Comment (michaelnebel) [inline csharp/ql/lib/change-notes/2023-05-09-models-as-data.md:4]: Maybe also update these to reflect the new sink kind names.

**Review by michaelnebel — APPROVED** [review]
Fantastic - thank you! :-)

**Comment by GeekMasher** [comment]
@jcogs33 @michaelnebel I have updated this PR now and pulled in the changes from https://github.com/github/codeql/pull/12916. Can you both review and let me know if anything else needs changing before merging?

**Review by michaelnebel — DISMISSED** [review]
Looks good to me!

**Review by jcogs33 — COMMENTED** [review]
I added a couple questions. 🙂

**Review by jcogs33 — DISMISSED** [review]
LGTM now. (assuming the CI checks all pass)

**Review by michaelnebel — APPROVED** [review]
LGTM 👍

**Review by jcogs33 — APPROVED** [review]

**Pull request merged**